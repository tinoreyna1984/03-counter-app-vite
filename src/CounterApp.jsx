import React, { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <br />
      <button onClick={handleAdd}>+</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubtract}>-</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};
