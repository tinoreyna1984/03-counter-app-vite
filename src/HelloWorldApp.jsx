import { CounterApp } from "./CounterApp";

export function HelloWorldApp() {
    const valor = 10;
    return (
      <>
        <h1>Hola mundo</h1>
      </>
    );
  }
  