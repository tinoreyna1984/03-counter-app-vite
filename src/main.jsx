import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
//import { HelloWorldApp } from "./HelloWorldApp";

const valor = 10;

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <CounterApp value={valor} />
  </React.StrictMode>
);
