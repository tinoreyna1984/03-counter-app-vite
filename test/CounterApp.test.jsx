import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Pruebas en CounterApp", () => {
  /**
   * Prueba 1: Comprobar que el valor del contador es el valor inicial
   */

  // valor de entrada
  const initialValue = 10;
  test("El componente debe hacer match con el snapshot", () => {
    // obtengo el valor contenedor del componente
    const { container } = render(<CounterApp value={initialValue} />);
    // comparo el valor contenedor con el snapshot
    expect(container).toMatchSnapshot();
  });

  /**
   * Prueba 2: CounterApp debe mostrar el valor de 100
   */
  test("El componente debe mostrar el valor de 100", () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText("100")).toBeTruthy();
  });

  /**
   * Prueba 3: CounterApp debe incrementar el valor del contador
   */
  test("El componente debe incrementar el valor del contador", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+"));
    expect(screen.getByText("11")).toBeTruthy(); // la prueba simula el incremento
  });

  /**
   * Prueba 4: CounterApp debe decrementar el valor del contador
   */
  test("El componente debe decrementar el valor del contador", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("-")); // la prueba simula ese decremento
    expect(screen.getByText("9")).toBeTruthy();
  });

  /**
   * Prueba 5
   */
  const newValue = 5;
  test("El componente debe decrementar el valor del contador", () => {
    render(<CounterApp value={newValue} />);
    fireEvent.click(screen.getByText("-")); // la prueba simula ese decremento
    expect(screen.getByText("4")).toBeTruthy();
  });

});
