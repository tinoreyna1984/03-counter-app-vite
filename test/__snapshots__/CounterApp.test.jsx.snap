// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp El componente debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <br />
  <button>
    +
  </button>
  <button>
    Reset
  </button>
  <button>
    -
  </button>
</div>
`;
